### ISSUE TITLE
BigNAS: Scaling Up Neural Architecture Search with Big Single-Stage Models

### ISSUE THREAD

**Issue by sile** [sile/readings#146]
URL
-----

https://arxiv.org/abs/2003.11142

Abstract
----------

> Neural architecture search (NAS) has shown promising results discovering models that are both accurate and fast. For NAS, training a one-shot model has become a popular strategy to rank the relative quality of different architectures (child models) using a single set of shared weights. However, while one-shot model weights can effectively rank different network architectures, the absolute accuracies from these shared weights are typically far below those obtained from stand-alone training. To compensate, existing methods assume that the weights must be retrained, finetuned, or otherwise post-processed after the search is completed. These steps significantly increase the compute requirements and complexity of the architecture search and model deployment. In this work, we propose BigNAS, an approach that challenges the conventional wisdom that post-processing of the weights is necessary to get good prediction accuracies. Without extra retraining or post-processing steps, we are able to train a single set of shared weights on ImageNet and use these weights to obtain child models whose sizes range from 200 to 1000 MFLOPs. Our discovered model family, BigNASModels, achieve top-1 accuracies ranging from 76.5% to 80.9%, surpassing state-of-the-art models in this range including EfficientNets and Once-for-All networks without extra retraining or post-processing. We present ablative study and analysis to further understand the proposed BigNASModels.

Table of Contents
--------------------

- (1) Introduction
- (2) Related Work
- (3) Architecture Search with Single-Stage Models
  - (3-1) Training a High-Quality Single-Stage Model
    - Sandwich Rule
    - Inplace Distillation
    - Initialization
    - Convergence Behavior
    - Regularization
    - Batch Norm Calibration
  - (3-2) Coarse-to-fine Architecture Selection
- (4) Experiments
  - (4-1) Search Space Definition
  - (4-2) Main Results on ImageNet
  - (4-3) Ablation Study
    - Ablation Study on Initialization
    - Ablation Study on Convergence Behavior
    - Ablation Study on Regularization
  - (4-4) Coarse-to-fine Architecture Selection
- (5) Analysis of BigNASModel
- (6) Conclusion
- (A) Architectures of BigNASModel
- (B) Implementation Details

BibTeX
------

```bibtex
@misc{yu2020bignas,
    title={BigNAS: Scaling Up Neural Architecture Search with Big Single-Stage Models},
    author={Jiahui Yu and Pengchong Jin and Hanxiao Liu and Gabriel Bender and Pieter-Jan Kindermans and Mingxing Tan and Thomas Huang and Xiaodan Song and Ruoming Pang and Quoc Le},
    year={2020},
    eprint={2003.11142},
    archivePrefix={arXiv},
    primaryClass={cs.CV}
}
```
